token/program-2022/tests/transfer_fee.rs

```diff
@@ -264,3 +265,234 @@ async fn fail_set_fee_unsupported_mint() {
         )))
     );
 }
+
+#[tokio::test]
+async fn set_transfer_fee_config_authority() {
+    let TransferFeeConfigWithKeypairs {
+        transfer_fee_config_authority,
+        withdraw_withheld_authority,
+        transfer_fee_config: TransferFeeConfig {
+            newer_transfer_fee, ..
+        },
+        ..
+    } = test_transfer_fee_config_with_keypairs();
+    let TestContext { token, .. } =
+        TestContext::new(vec![ExtensionInitializationParams::TransferFeeConfig {
+            transfer_fee_config_authority: transfer_fee_config_authority.pubkey().into(),
+            withdraw_withheld_authority: withdraw_withheld_authority.pubkey().into(),
+            transfer_fee_basis_points: newer_transfer_fee.transfer_fee_basis_points.into(),
+            maximum_fee: newer_transfer_fee.maximum_fee.into(),
+        }])
+        .await
+        .unwrap();
+
+    let new_authority = Keypair::new();
+    let wrong = Keypair::new();
+
+    // fail, wrong signer
+    let err = token
+        .set_authority(
+            token.get_address(),
+            Some(&new_authority.pubkey()),
+            instruction::AuthorityType::TransferFeeConfig,
+            &wrong,
+        )
+        .await
+        .unwrap_err();
+    assert_eq!(
+        err,
+        TokenClientError::Client(Box::new(TransportError::TransactionError(
+            TransactionError::InstructionError(
+                0,
+                InstructionError::Custom(TokenError::OwnerMismatch as u32)
+            )
+        )))
+    );
+
+    // success
+    token
+        .set_authority(
+            token.get_address(),
+            Some(&new_authority.pubkey()),
+            instruction::AuthorityType::TransferFeeConfig,
+            &transfer_fee_config_authority,
+        )
+        .await
+        .unwrap();
+    let state = token.get_mint_info().await.unwrap();
+    let extension = state.get_extension::<TransferFeeConfig>().unwrap();
+    assert_eq!(
+        extension.transfer_fee_config_authority,
+        Some(new_authority.pubkey()).try_into().unwrap(),
+    );
+
+    // assert new_authority can update transfer fee config, and old cannot
+    let transfer_fee_basis_points = u16::MAX;
+    let maximum_fee = u64::MAX;
+    token
+        .set_transfer_fee(
+            &transfer_fee_config_authority,
+            transfer_fee_basis_points,
+            maximum_fee,
+        )
+        .await
+        .unwrap_err();
```

Comment:
nit: Could we check the exact error type here?

Reply:
We could. I didn't think it was worth the effort, but I'll do it for you 😅

Reply:
It's just a nit, so feel free to leave it if it's annoying